### ISSUE TITLE
setter on Labels.properties does not accept default value of Labels.properties

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#2477]
Playing with some serialisation over on napari animation and ran into an issue with setting `Labels.properties`

example:

```python
import napari
from skimage import data

import scipy.ndimage as ndi

blobs = data.binary_blobs(length=128, volume_fraction=0.1, n_dim=3)
viewer = napari.Viewer(ndisplay=3)
blobs_layer = viewer.add_image(blobs.astype(float), name='blobs')
labels = ndi.label(blobs)[0]
labels_layer = viewer.add_labels(labels, name='blob ID')

labels_layer.properties = labels_layer.properties
```

gives

```python
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/napari-animation/lib/python3.9/site-packages/IPython/core/interactiveshell.py", line 3427, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-2-39c90256163c>", line 1, in <module>
    runfile('/Users/aburt/Programming/napari-animation/playground/labels_properties_setter_issue.py', wdir='/Users/aburt/Programming/napari-animation/playground')
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_bundle/pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/Applications/PyCharm.app/Contents/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/Users/aburt/Programming/napari-animation/playground/labels_properties_setter_issue.py", line 12, in <module>
    labels_layer.properties = labels_layer.properties
  File "/Users/aburt/Programming/napari/napari/layers/labels/labels.py", line 338, in properties
    self._label_index = label_index
UnboundLocalError: local variable 'label_index' referenced before assignment
```

will submit a PR to fix now

**Comment by sofroniewn** [comment]
Can this be closed @alisterburt? @Czaki might be looking into something similar in #2657 too, so maybe it can be closed after that?

**Comment by alisterburt** [comment]
Yep this is closed by #2478, looks like the other issue is a little different
